File: Ruby/TP8/Ej5/checking_account.rb
require_relative 'account'
class CheckingAccount < Account

  def initialize(id, overdraft)
    super(id)
    @overdraft=overdraft
    @balance = 0
  end

  def can_extract(amount)
    @balance+@overdraft-amount>=0
  end

  def get_overdraft
    @overdraft
  end

end

File: Java/SegundoParcial/TP6/Ej4/CheckingAccount.java
package ar.itba.edu.POO.SegundoParcial.TP6.Ej4;

public class CheckingAccount extends BankAccount {

    private final double overdraft;

    public CheckingAccount(int id, double overdraft) {
        super(id);
        this.overdraft=overdraft;
    }

    @Override
    protected boolean canExtract(double amount) {
        return balance+overdraft-amount>=0;
    }

    public double getOverdraft(){
        return overdraft;
    }
}
